ecs-init/docker/dependencies.go

```diff
@@ -96,3 +129,17 @@ var standardFS = &_standardFS{}
 func (s *_standardFS) ReadFile(filename string) ([]byte, error) {
 	return ioutil.ReadFile(filename)
 }
+
+func isNetworkError(err error) bool {
+	_, ok := err.(*net.OpError)
+	return ok
+}
+
+func isRetryablePingError(err error) bool {
+	godockerError, ok := err.(*godocker.Error)
+	if ok && godockerError.Status != http.StatusOK {
```

Comment:
If you are checking against [this line](https://github.com/fsouza/go-dockerclient/blob/master/client.go#L392), is there any chance the `godockerError.Status == http.StatusOK`? Since it has been checked [here](https://github.com/fsouza/go-dockerclient/blob/master/client.go#L392).